lib/vscode/src/vs/platform/extensionManagement/common/extensionGalleryService.ts

```diff
@@ -543,43 +548,53 @@ export class ExtensionGalleryService implements IExtensionGalleryService {
 			'Content-Length': String(data.length)
 		};
 
-		const context = await this.requestService.request({
-			type: 'POST',
-			url: this.api('/extensionquery'),
-			data,
-			headers
-		}, token);
+		const galleryExtensions: IRawGalleryExtension[] = [];
+		const seenExtensions: Set<String> = new Set();
 
-		if (context.res.statusCode && context.res.statusCode >= 400 && context.res.statusCode < 500) {
-			return { galleryExtensions: [], total: 0 };
-		}
+		for (const url of this.api('/extensionquery')) {
+			const context = await this.requestService.request({
+				type: 'POST',
+				url: url,
+				data,
+				headers
+			}, token);
 
-		const result = await asJson<IRawGalleryQueryResult>(context);
-		if (result) {
-			const r = result.results[0];
-			const galleryExtensions = r.extensions;
-			const resultCount = r.resultMetadata && r.resultMetadata.filter(m => m.metadataType === 'ResultCount')[0];
-			const total = resultCount && resultCount.metadataItems.filter(i => i.name === 'TotalCount')[0].count || 0;
+			if (context.res.statusCode && context.res.statusCode >= 400 && context.res.statusCode < 500) {
+				return { galleryExtensions: [], total: 0 };
+			}
 
-			return { galleryExtensions, total };
+			const result = await asJson<IRawGalleryQueryResult>(context);
+			if (result) {
+				const r = result.results[0];
+				for (const extension of r.extensions) {
+					if (!seenExtensions.has(extension.extensionId)) {
+						galleryExtensions.push(extension);
+						seenExtensions.add(extension.extensionId);
+					}
+				}
+			}
 		}
-		return { galleryExtensions: [], total: 0 };
+		return { galleryExtensions, total: galleryExtensions.length };
 	}
 
 	async reportStatistic(publisher: string, name: string, version: string, type: StatisticType): Promise<void> {
+		// TODO: investigate further - currently we just send stats everywhere
```

Comment:
I know folks have GitLens and tools to know who left a TODO comment but at least this puts your name right there so we don't forget. 

```suggestion
		// TODO(oxy): investigate further - currently we just send stats everywhere
```

Reply:
:+1: will fix!